renderer/reducers/lnd.js

```diff
@@ -429,16 +430,40 @@ export const recoverOldWallet = () => async (dispatch, getState) => {
     // Start lnd with the provided wallet config.
     await dispatch(startLnd(wallet))
 
+    // Set up a listener that resolves once the lightning interface has become active.
+    // This is the point where we really know that the wallet has been created and we can connect to it.
+    const waitForLightning = new Promise(resolve => {
+      handleLightningActive = proxyValue(() => resolve())
+      grpc.prependListener('GRPC_LIGHTNING_SERVICE_ACTIVE', handleLightningActive)
+    })
```

Comment:
This looks like a fairly low-level routine and also like a reusable one. Consider extracting this into util.

Reply:
I'e extracted this out into a helper